packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts

```diff
@@ -48,13 +48,24 @@ export const useInfoLabel_unstable = (props: InfoLabelProps, ref: React.Ref<HTML
     required: !!info,
     defaultProps: {
       id: useId('infobutton-'),
-      info,
       size,
+      info,
+    },
+  });
+
+  const infoButtonInfo = resolveShorthand(infoButton?.info, {
+    defaultProps: {
+      id: useId('infobutton-info-'),
     },
   });
 
   if (infoButton) {
+    infoButton.info = infoButtonInfo;
```

Comment:
nit: If you take my `baseId` suggestion, then this call to resolveShorthand can move inside the `if (infoButton)`:
```suggestion
  if (infoButton) {
    infoButton.info = resolveShorthand(infoButton.info, {
      defaultProps: {
        id: baseId + '__info',
      },
    });
```